EL-2428 - Adding Chart.js Directives

ashley-hunter opened the pull request:
Packages added & updated - npm install required.

Added Chart.js options for most of our flot charts using `ng2-charts` external library.

There are a few small differences between new charts and old charts due to how the different libraries work, however for the most part they are very similar.

There are a few charts that are missing features, for example stacked line chart does not show forecast data, and timeline chart which doesn't have a chart.js equivalent for now. These will require us to write Chart.js plugins to provide this functionality which I will raise other jiras for.

Documentation for each chart is fairly minimal and virtually the same for each chart, I have linked to all the external libs and documentation sites in each section for full info on them as there is just far too much to event attempt to cover in our doc site.

donna-marijne commented on docs/app/pages/charts/sections/bar-charts/bar-chart/bar-chart.component.html:
```diff
@@ -0,0 +1,124 @@
+
+<p>
+    Bar Charts can be added using the <code>ng2-charts</code> library. 
+    The <code>chart.js</code> libray needs to be imported and the <code>ChartsModule</code> needs to be added to the appropriate NgModule.
```
Typo 'libray'

donna-marijne commented on docs/app/pages/charts/sections/bar-charts/bar-chart/bar-chart.component.html:
```diff
@@ -0,0 +1,124 @@
+    <p class="m-b-nil">
+        <b>Documentation: </b>
+        <a class="dependency-link" href="http://valor-software.com/ng2-charts/">Angular Directive Documentation</a>, 
+        <a class="dependency-link" href="http://www.chartjs.org/docs/">Chart Library Documentation</a>
```
These hyperlinks are not visually distinctive, maybe use standard text and hyperlink styles here.

donna-marijne commented on docs/app/pages/charts/sections/bar-charts/horizontal-bar-chart/horizontal-bar-chart.component.html:
```diff
@@ -0,0 +1,129 @@
+
+<p>
+    Horizontal Bar Charts can be added using the <code>ng2-charts</code> library. 
+    The <code>chart.js</code> libray needs to be imported and the <code>ChartsModule</code> needs to be added to the appropriate NgModule.
```
Typo 'libray'

donna-marijne commented on docs/app/pages/charts/sections/bar-charts/horizontal-bar-chart/horizontal-bar-chart.component.html:
```diff
@@ -0,0 +1,129 @@
+            [colors]="barChartColors"
+            chartType="horizontalBar">
+    </canvas>
+
```
Looks a lot better than the old one :) Is there any way to have the x-axis line along the bottom instead of the top, with the labels?

donna-marijne commented on docs/app/pages/charts/sections/bar-charts/stacked-bar-chart/stacked-bar-chart.component.html:
```diff
@@ -0,0 +1,129 @@
+
+<p>
+    Stacked Bar Charts can be added using the <code>ng2-charts</code> library. 
+    The <code>chart.js</code> libray needs to be imported and the <code>ChartsModule</code> needs to be added to the appropriate NgModule.
```
Typo 'libray'

donna-marijne commented on docs/app/pages/charts/sections/scrollable-chart/scrollable-chart/scrollable-chart.component.html:
```diff
@@ -0,0 +1,137 @@
+<br>
+
+<p>The <code>baseChart</code> directive should be added to a <code>canvas</code> element and the following attributes can be used to customize the chart behavior and appearance:</p>
+
```
Some more explanation might be needed here since scrolling isn't provided by the ng2-charts but rather provided through custom code.

ashley-hunter commented:
Fixed typos, added description for scrollable charts, reduced horizontal scrolling where possible (a few places were using string templating and couldn't really split onto multiple lines. Issue with horizontal bar chart x-axis will require more investigation. Will cover as part of the ticket to implement the labels above bars in horizontal charts. Hyperlinks changed
